sympy/solvers/diophantine.py

```diff
@@ -152,11 +152,24 @@ def diophantine(eq, param=symbols("t", integer=True), syms=None):
             if not is_sequence(syms):
                 raise TypeError(
                     'syms should be given as a sequence, e.g. a list')
-            syms = [i for i in syms if i in var]
+
             if syms != var:
-                map = dict(zip(syms, range(len(syms))))
-                return set([tuple([t[map[i]] for i in var])
-                    for t in diophantine(eq, param)])
+                extra_syms = list(filter(lambda x: x not in var, syms))
+                syms_var = list(filter(lambda x: x in syms, var))
+                map = dict(zip(syms_var, range(len(syms_var))))
+                soln = diophantine(eq, param)
+
+                if len(syms) > len(var):
+                    num_sym = numbered_symbols("m", integer=True, start=1)
```

Comment:
I followed [this way](https://github.com/Shekharrajak/sympy/blob/e191a52314de2c4ca1d01c072d1ce7910c0c9f64/sympy/solvers/diophantine.py#L260). It seems Dummy should be there.